### ISSUE TITLE
Change the number of GPU's while continuing training ? 

### ISSUE THREAD

**Issue by manchandasahil** [microsoft/DeepSpeed#787]
I can see that the Optimizer states are saved per rank when i train a bert model in azure. Hence if i train with 8 gpus there are 8 optimizer state files. 
Does that mean when i stop and continue training, it is not possible to change the number of GPU's ?

**Comment by tjruwase** [comment]
The per-rank optimizer state is because ZeRO partitions optimizer states across ranks to save memory. We do provide support for changing GPU counts for continued training from restored checkpoints. Unfortunately, we are currently debugging an issue with this support that should be resolved soon.

**Comment by sdtblck** [comment]
Hi @tjruwase any updates on this? You say you provide support for it - can you point me toward said support? I have a training run I started on 64 GPUs, and am now trying to change to fewer, but running into problems relating to this.

**Comment by tjruwase** [comment]
@sdtblck, unfortunately the investigation into the elastic checkpoint support was stalled. Can you tell me which ZeRO stage you are using?

**Comment by sdtblck** [comment]
Hi @tjruwase it's ZeRO stage 1. I'm assuming it's something to do with `optimizer.elastic_checkpoint`? I see it's true by default, but when loading on a different number of machines the loss is rocketing up really high when we load a checkpoint.

**Comment by sdtblck** [comment]
do you have any idea what the problem is more specifically?

**Comment by zhoubay** [comment]
I also want to know how to change number of GPUs while continuing training, are there any solutions for now? I'm using ZeRO Stage 2 in pytorch lightning. @tjruwase @sdtblck

**Comment by dongshou** [comment]
me too when using ZeRO Stage 2

**Comment by tjruwase** [comment]
@zhoubay, @dongshou this is now supported through our Universal Checkpointing feature
https://github.com/microsoft/DeepSpeed/blob/master/blogs/deepspeed-ucp/README.md
